controllers/nodefeaturediscovery_controller.go

```diff
@@ -19,60 +19,58 @@ package controllers
 import (
 	"context"
 
-	"github.com/go-logr/logr"
 	appsv1 "k8s.io/api/apps/v1"
 	corev1 "k8s.io/api/core/v1"
 	k8serrors "k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	"k8s.io/client-go/tools/record"
-	"k8s.io/klog"
+	"k8s.io/klog/v2"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/controller-runtime/pkg/event"
-	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	nfdv1 "github.com/kubernetes-sigs/node-feature-discovery-operator/api/v1"
 )
 
-// log is used to set the logger with a name that describes the actions of
-// functions and arguments in this file
-var log = logf.Log.WithName("controller_nodefeaturediscovery")
-
 // nfd is an NFD object that will be used to initialize the NFD operator
 var nfd NFD
 
-// NodeFeatureDiscoveryReconciler reconciles a NodeFeatureDiscovery object
-type NodeFeatureDiscoveryReconciler struct {
+// NodeFeatureDiscoveryLogger is a dummy logger struct that is used with
+// the NodeFeatureDiscoveryReconciler to initiate a logger.
+type NodeFeatureDiscoveryLogger struct {
+}
```

Comment:
I don't quite get the usefulness of this? Why not just add logger methods directly to `NodeFeatureDiscoveryReconciler`? As it doesn't really implement any logger interface (known to me) I'd suggest to scrap the logging methods and do simply`s/logger/klog/`. WDYT?